### PR TITLE
Converge metric tags

### PR DESCRIPTION
**Pull request by acrmp**
Thank you for submitting a pull request to the rep repository!

We appreciate the contribution. To help us understand the context for your pull request, please fill out this template to the best of your ability.

## Please make sure to complete all of the following steps

1. Check the [Contributing document](https://github.com/cloudfoundry/diego-release/blob/develop/CONTRIBUTING.md) on how to sign the CLA and run tests.
1. Submit your PR to this repo.
1. ~~[**Submit an accompanying PR Review Request**](https://github.com/cloudfoundry/diego-release/issues/new?assignees=&labels=&template=pr-review-request.md&title=%5BREP+PR+REVIEW%5D%3A) referencing this PR so the Diego Team knows to review your pull request.~~ 
* ~~**Note: this PR will not be reviewed unless you submit the [PR Review Request](https://github.com/cloudfoundry/diego-release/issues/new?assignees=&labels=&template=pr-review-request.md&title=%5BREP+PR+REVIEW%5D%3A)**.~~

Not submitting a separate PR Review Request as we already have https://github.com/cloudfoundry/diego-release/issues/662 as a central issue to link PRs to.

***************************

### What is this change about?

This PR modifies the rep to pass metric tags back to the BBS when calling `StartActualLRP` or `EvacuateRunningActualLRP`.

It introduces a new version of the rep update lrp instance endpoint in order to avoid setting internal routes to nil when different versions of the bbs and rep are present.  @ctlong and @rroberts2222 chatted with @mariash about how to express this in the endpoint path and agreed to express it as a prefix to the URL rather than for example the BBS approach of a revision suffix.

### What problem it is trying to solve?

Ensuring that metric tags are converged when the initial attempt to update the lrp fails.

### What is the impact if the change is not made?

Metric tags may not be applied.

### How should this change be described in diego-release release notes?

It shouldn't be described individually, only as part of the larger change.

### Please provide any contextual information.

https://github.com/cloudfoundry/diego-release/issues/662
https://github.com/cloudfoundry/bbs/pull/56
https://github.com/cloudfoundry/route-emitter/pull/19

### Tag your pair, your PM, and/or team!

@rroberts2222 @ctlong @Benjamintf1

Thank you!

### CODE REVIEW

**Comment by mariash** [comment]
I see both routes call the same handler. Will it be a separate change? v0 and v1 handlers?

**Comment by ctlong** [comment]
@mariash difference handlers weren't required to solve the edge case we were concerned about, so we didn't add them here.

❓ Do you want the routes to have different handlers?

I agree it's cleaner from a backward compatibility perspective. If y'all would look to rip out that functionality sooner rather than later it's just more code to delete though.

File: client.go [hunk 1]
```diff
@@ -295,6 +296,63 @@ func (c *client) UpdateLRPInstance(
 
 	// We are assuming that a 404 means that the rep has not updated
 	// yet, but will roll soon.  This is for backwards compatibility.
+	if resp.StatusCode == http.StatusNotFound {
+		logger.Error("failed-with-status", err, lager.Data{"status-code": resp.StatusCode, "msg": http.StatusText(resp.StatusCode)})
+		// The v1 UpdateLRPInstance route is only for updating InternalRoutes
+		// on old versions of rep. This is for backwards compatibility.
+		if update.InternalRoutes != nil {
```
Comment (ctlong) [inline client.go:303]: This is the change that really prevents overwriting internal routes in the `new bbs, old rep` case.

Comment (mariash) [reply client.go:303]: Makes sense, just saw the implementation of [executor.Update](https://github.com/cloudfoundry/executor/blob/update-metric-tags/depot/containerstore/storenode.go#L598-L619) that it handles both cases.

**Review by mariash — APPROVED** [review]

**Review by mariash — APPROVED** [review]

**Pull request merged**